dcpy/connectors/edm/recipes.py

```diff
@@ -178,13 +178,6 @@ def import_dataset(
 
     if ds.version == "latest" or ds.version is None:
         raise Exception(f"Cannot import a dataset without a resolved version: {ds}")
-    if ds.preprocessor is not None and ds.file_type not in (
-        DatasetType.csv,
-        DatasetType.parquet,
-    ):
-        raise Exception(
-            "Preprocessor can only be specified for csv or parquet datasets"
-        )
```

Comment:
fixes FACDB build failing [on main](https://github.com/NYCPlanning/data-engineering/actions/runs/7042259765/job/19166139098#step:8:238) and it was failing here

this exception doesn't seem raising since the `if/elif` logic below doesn't try to import a preprocessor python module if the dataset is a pgdump

Reply:
huh, I'm confused about why this would be happening now. But this dataset should be pg_dump, so the recipe is correct. 
```
│ │                ds = Dataset(                                             │ │
│ │                     │   name='dcp_facilities_with_unmapped',             │ │
│ │                     │   version='20230430',                              │ │
│ │                     │   version_env_var='VERSION_PREV',                  │ │
│ │                     │   import_as=None,                                  │ │
│ │                     │   file_type='pg_dump',                             │ │
│ │                     │   preprocessor=DataPreprocessor(                   │ │
│ │                     │   │   module='facdb.pipelines',                    │ │
│ │                     │   │   function='dispatch'                          │ │
│ │                     │   )                                                │ │
│ │                     )    
```
I think it's fine to remove the bit that raises the exception. Maybe just add a `logging.warn` in that case?

Reply:
yea the recipe is correct. just seems like when a default preprocessor is defined, we wouldn't wanna raise an exception for sources that can't/would never use it

Reply:
Alex, what's the rationale for warning? Like is there a risk in your mind of a pg_dump preprocessing step? Just a bit unfamiliar with what's actually happening with these preprocessors

Reply:
Or is a preprocessor just not actually being applied in the case of the pg_dump file type

Reply:
it's not applied to any pg_dump files. a warning does seem nice as breadcrumbs in case someone expects their default preprocessor to apply to every source dataset